slurm/slurm.md

```diff
@@ -0,0 +1,941 @@
+KillWait=30
+MessageTimeout=10
+ResvOverRun=0
+MinJobAge=300
+OverTimeLimit=0
+SlurmctldTimeout=120
+SlurmdTimeout=300
+UnkillableStepTimeout=60
+VSizeFactor=0
+Waittime=0
+#
+
+#Job Scheduling
+SchedulerTimeSlice=30
+SchedulerType=sched/backfill
+SchedulerParameters=nohold_on_prolog_fail
+SelectType=select/cons_res
+SelectTypeParameters=CR_Core
+
+# Job Priority
+PriorityType=priority/multifactor
+PriorityFlags=FAIR_TREE
+## 2 week half-life
+PriorityDecayHalfLife=14-0
+# The larger the job, the greater its job size priority.
+PriorityFavorSmall=NO
+# The job's age factor reaches 1.0 after waiting in the
+# queue for 2 weeks.
+PriorityMaxAge=14-0
+# This next group determines the weighting of each of the
+# components of the Multi-factor Job Priority Plugin.
+# The default value for each of the following is 1.
+PriorityWeightAge=1000
+PriorityWeightJobSize=1000
+PriorityWeightPartition=10000000
+PriorityWeightQOS=10000000
+PriorityWeightTRES=CPU=2000,Mem=4000
+PriorityWeightFairshare=100000
+#
+
+# LOGGING AND ACCOUNTING
+AccountingStorageType=accounting_storage/slurmdbd
+AccountingStorageHost=X.X.X.X
+#AccountingStoreFlags=job_comment
+AccountingStoragePort=6819
+AccountingStorageEnforce=limits,associations,qos
+AccountingStorageTRES=CPU,Mem
+SlurmctldLogFile=/var/log/slurmctld.log
+SlurmdLogFile=/var/log/slurmd.log
+JobAcctGatherFrequency=30
+JobAcctGatherType=jobacct_gather/cgroup
+SlurmctldDebug=info
+SlurmdDebug=info
+SlurmSchedLogFile=/var/log/slurmSched.log
+SlurmSchedLogLevel=1
+#
+#Energy consumption test
+AcctGatherEnergyType=acct_gather_energy/rapl
+AcctGatherNodeFreq=1
+#
+
+#Partitions
+PartitionName=HPC                       MaxTime=UNLIMITED Nodes=wn001  MaxNodes=1  OverSubscribe=YES AllowGRoups=hpc
+
+#COMPUTE NODES
+NodeName=wn001 NodeAddr=X.X.X.X  CPUs=2 Sockets=1 CoresPerSocket=2 ThreadsPerCore=1 State=IDLE RealMemory=2048
+
+
+```
+
+Create a /etc/slurm/cgroup.conf
+
+```
+CgroupAutomount=yes
+ConstrainCores=yes
+ConstrainRAMSpace=yes
+ConstrainSwapSpace=yes
+TaskAffinity=no
+
+```
+
+Enable and start the slurmctld:
+
+```
+systemctl enable slurmctld
+systemctl start slurmctld
+```
+
+##### NOTE: Check that the slurmdbd service is running before starting the slurmctld:
+
+### On the `shadow.local` machine
+
+Copy slurm.conf  and cgroup.conf to `shadow.local`  from `master.local` machine:
+
+```
+scp /etc/slurm/slurm.conf root@shadow.local:/etc/slurm/.
+
+scp /etc/slurm/cgroup.conf root@shadow.local:/etc/slurm/.
+
+```
+
+Enable and start the slurmctld:
+
+```
+systemctl enable slurmctld
+systemctl start slurmctld
+```
+
+
+### On the `wn001.local`(worker node) machine
+
+Setup user munge and slurm
+
+```
+export MUNGEUSER=981
+groupadd -g $MUNGEUSER munge
+useradd  -m -c "MUNGE Uid 'N' Gid Emporium" -d /var/lib/munge -u $MUNGEUSER -g munge  -s /sbin/nologin munge
+export SlurmUSER=982
+groupadd -g $SlurmUSER slurm
+useradd  -m -c "Slurm workload manager" -d /var/lib/slurm -u $SlurmUSER -g slurm  -s /bin/bash slurm
```

Comment:
I think groups and users should be system groups (add `-r`).
Slurm user should have `/sbin/nologin` shell, too.

Reply:
You are completely right.